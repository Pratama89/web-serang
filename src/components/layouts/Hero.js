import Image from "next/image";
import Right from "../icons/Right";
import Link from "next/link";
import { Carousel } from "flowbite-react";

export default function Hero() {
  return (
    <section
      className="hero  pt-40 bg-[url('/laminating-hero.jpg')] flex justify-center items-center p-20 bg-cover xs:bg-[url('/latar-depan.png')] xs:bg-cover xs:w-screen xs:h-screen xs:-mt-12 xs:pt-40 xs:flex xs:justify-center xs:p-10	"
      id="home">
      
        <div className="py-12 xs:col-span-2 xs:flex xs:flex-col ">
          <h1 className="text-4xl text-black xs:text-3xl font-semibold xs:text-center">
            <span className="text-primary">Mutiara</span> Laminating
          </h1>
          <p className="mt-4 text-black text-lg  xs:text-center xs:text-white">
            Dipercaya oleh lebih dari ratusan pelaku industri sebagai penyedia
            jasa laminating kain dan busa sejak tahun 2014.
          </p>
          <div className="flex justify-start gap-4 items-center mt-8 text-sm xs:flex-col">
            <button className="bg-primary flex gap-2 text-white px-4 py-2 rounded-full ">
              Layanan Laminating
              <Right />
            </button>
            <Link href={"https:wa.link/h7ywr2"}>
              <button className="flex justify-center items-center gap-2  text-white text-sm px-4 py-2 rounded-full xs:px-6 xs:py-4 ">
                Kontak Admin
                <Right />
              </button>
            </Link>
          </div>
        </div>
        <Image
          src={"/img/mesin-laminating2.png"}
          alt="hero image"
          width={600}
          height={300}
          className="xs:hidden "
        />
      
    </section>
  );
}
