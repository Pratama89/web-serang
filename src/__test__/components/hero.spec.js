import Hero from "@/components/layouts/Hero";
import { render, screen } from "@testing-library/react";
import { describe } from "node:test";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/hero",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}));

describe("Hero", () => {
  it("should render", () => {
    const page = render(<Hero />);
    // expect(screen.getByTestId("title").textContent).toBe("Header");
    expect(page).toMatchSnapshot();
  });
});
