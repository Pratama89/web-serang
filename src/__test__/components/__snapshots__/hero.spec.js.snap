// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="hero  items-center mt-24"
        id="home"
      >
        <div
          class="py-12 xs:col-span-2 xs:flex xs:flex-col xs:items-center "
        >
          <h1
            class="text-4xl font-semibold "
          >
            <span
              class="text-primary "
            >
              Mutiara
            </span>
             Laminating
          </h1>
          <p
            class="mt-4 text-gray-400 text-sm xs:text-center"
          >
            Dipercaya oleh lebih dari ratusan pelaku industri sebagai penyedia jasa laminating kain dan busa sejak tahun 2014.
          </p>
          <div
            class="flex justify-start gap-4 items-center mt-8 text-sm"
          >
            <a
              href="#contact"
            >
              <button
                class="flex gap-2 bg-primary text-white text-sm px-4 py-2 rounded-full"
              >
                Kontak Admin
                <svg
                  class="w-6 h-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </a>
          </div>
        </div>
        <img
          alt="hero image"
          class="xs:hidden  "
          data-nimg="1"
          decoding="async"
          height="400"
          loading="lazy"
          src="/_next/image?url=%2Fmesin-laminating.png&w=1920&q=75"
          srcset="/_next/image?url=%2Fmesin-laminating.png&w=828&q=75 1x, /_next/image?url=%2Fmesin-laminating.png&w=1920&q=75 2x"
          style="color: transparent;"
          width="800"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="hero  items-center mt-24"
      id="home"
    >
      <div
        class="py-12 xs:col-span-2 xs:flex xs:flex-col xs:items-center "
      >
        <h1
          class="text-4xl font-semibold "
        >
          <span
            class="text-primary "
          >
            Mutiara
          </span>
           Laminating
        </h1>
        <p
          class="mt-4 text-gray-400 text-sm xs:text-center"
        >
          Dipercaya oleh lebih dari ratusan pelaku industri sebagai penyedia jasa laminating kain dan busa sejak tahun 2014.
        </p>
        <div
          class="flex justify-start gap-4 items-center mt-8 text-sm"
        >
          <a
            href="#contact"
          >
            <button
              class="flex gap-2 bg-primary text-white text-sm px-4 py-2 rounded-full"
            >
              Kontak Admin
              <svg
                class="w-6 h-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
      <img
        alt="hero image"
        class="xs:hidden  "
        data-nimg="1"
        decoding="async"
        height="400"
        loading="lazy"
        src="/_next/image?url=%2Fmesin-laminating.png&w=1920&q=75"
        srcset="/_next/image?url=%2Fmesin-laminating.png&w=828&q=75 1x, /_next/image?url=%2Fmesin-laminating.png&w=1920&q=75 2x"
        style="color: transparent;"
        width="800"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
